lib/classifier-reborn/backends/bayes_memory_backend.rb

```diff
@@ -0,0 +1,72 @@
+# Author: Sawood Alam <@ibnesayeed>
+
+module ClassifierReborn
+  class BayesMemoryBackend
+    def initialize
+      @total_words     = 0
+      @total_trainings = 0
+      @category_counts = Hash.new { |h, k| h[k] = {training: 0, word: 0} }
+      @categories      = {}
+    end
+
+    def total_words
+      @total_words
+    end
+
+    def update_total_words(diff)
+      @total_words += diff
+    end
+
+    def total_trainings
```

Comment:
@Ch4s3, in the Redis backend they are not plain in-memory attributes. They are stored in the Redis instance. Keeping them in memory as plain attributes would require to traverse through all the records and when initialized from an existing dataset. This was the whole point of using another backend to keep the counters and other state information in the backend store while keeping the configurations and other similar things in the memory.